clippy_lints/src/utils/conf.rs

```diff
@@ -205,10 +205,33 @@ macro_rules! define_Conf {
 }
 
 define_Conf! {
-    /// Lint: Arithmetic.
+    /// Lint: arithmetic_side_effects.
     ///
-    /// Suppress checking of the passed type names.
-    (arithmetic_side_effects_allowed: rustc_data_structures::fx::FxHashSet<String> = <_>::default()),
+    /// Suppress checking of the passed type pair names grouped by commas in binary operations like
+    /// addition or multiplication.
+    ///
+    /// Supports the "*" wildcard to indicate that a certain type won't trigger the lint regardless
+    /// of the involved counterpart. For example, `SomeType,*` or `*,AnotherType`.
+    ///
+    /// Pairs are asymmetric, which means that `SomeType,AnotherType` is not the same as
+    /// `AnotherType,SomeType`.
+    ///
+    /// # Example
+    ///
+    /// ```toml
+    /// arithmetic-side-effects-allowed-bin = ["SomeType,f32", "AnotherType,*"]
+    /// ```
+    (arithmetic_side_effects_allowed_bin: Vec<String> = <_>::default()),
```

Comment:
Would this work with `Vec<[String; 2]>`? `[["SomeType", "f32"], ..]` would be nicer than splitting on an internal comma IMO

Reply:
Surprisingly (at least for me), it is possible. Changed.